arctic/chunkstore/chunkstore.py

```diff
@@ -262,22 +262,28 @@ def read(self, symbol, chunk_range=None, filter_data=True, **kwargs):
         if chunk_range is not None:
             spec.update(chunker.to_mongo(chunk_range))
 
-        by_start_segment = [(SYMBOL, pymongo.ASCENDING),
-                            (START, pymongo.ASCENDING),
-                            (SEGMENT, pymongo.ASCENDING)]
-        segment_cursor = self._collection.find(spec, sort=by_start_segment)
-
         chunks = defaultdict(list)
-        for _, segments in groupby(segment_cursor, key=lambda x: (x[START], x[SYMBOL])):
-            segments = list(segments)
-            mdata = self._mdata.find_one({SYMBOL: segments[0][SYMBOL],
-                                          START: segments[0][START],
-                                          END: segments[0][END]})
-
-            # when len(segments) == 1, this is essentially a no-op
-            # otherwise, take all segments and reassemble the data to one chunk
-            chunk_data = b''.join([doc[DATA] for doc in segments])
-            chunks[segments[0][SYMBOL]].append({DATA: chunk_data, METADATA: mdata})
+        joined = self._collection.aggregate([
+            {'$match': spec},
+            {'$lookup': {
+                'from': self._mdata.name,
+                'let': {'symbol': '${}'.format(SYMBOL),
+                        'start': '${}'.format(START)},
+                'pipeline': [
+                    {'$match': {
+                        '$expr': {
+                            '$and': [{'$eq': ['${}'.format(SYMBOL), '$$symbol']},
+                                     {'$eq': ['${}'.format(START), '$$start']}]}}},
+                    {'$project': {'_id': 0, 'm': 0}}], 
+                'as': METADATA}},
+            {'$sort': {START: 1,
+                       SEGMENT: 1}}],
+        allowDiskUse=True)
```

Comment:
Would be useful to have some comments on what the pipeline is doing.  Particularly if we're spilling to disk would be good to understand why this is needed?

Reply:
Updated with comments. The aggregation pipeline has a memory limit for each step so in order to enable it for large datasets, we set the allowDiskUse flag. 

https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/#example-aggregate-method-external-sort